### PR TITLE
Add/disable on steering input

### PR DESCRIPTION
**Pull request by evanlivingston**
This branch adds global control disable if the steering wheel torque rises above a threshold for a specific period of time, using a low pass filter.

### CODE REVIEW

File: firmware/steering/kia_soul_ps/steering_control_module.ino [hunk 1]
```diff
@@ -386,6 +426,9 @@ static void process_ps_ctrl_steering_command( const uint8_t * const rx_frame_buf
     current_ctrl_state.commanded_steering_angle = control_data->steering_wheel_angle_command / 9.0 ;
 
     current_ctrl_state.steering_angle_rate_max = control_data->steering_wheel_max_velocity * 9.0;
+    current_ctrl_state.steering_angle_rate_max = 350;
```
Comment (LucasBuckland) [inline firmware/steering/kia_soul_ps/steering_control_module.ino]: Should we be setting this manually for users? Is this necessary for steering wheel detector to work?

File: firmware/steering/kia_soul_ps/steering_control_module.ino [hunk 2]
```diff
@@ -266,14 +301,17 @@ void calculateTorqueSpoof( float torque, uint16_t * TSpoofL, uint16_t * TSpoofH
 // A function to parse incoming serial bytes
 void processSerialByte( uint8_t incomingSerialByte)
```
Comment (LucasBuckland) [inline firmware/steering/kia_soul_ps/steering_control_module.ino]: We removed this function in the main branch I think. Nbd as Petes PR should remove it, but just noting it.

File: firmware/steering/kia_soul_ps/steering_control_module.ino [hunk 3]
```diff
@@ -89,6 +88,11 @@
 // static structures
 // *****************************************************
 
+float filterFrequency = 0.07;
```
Comment (LucasBuckland) [inline firmware/steering/kia_soul_ps/steering_control_module.ino]: Might be cleaner as a macro

File: firmware/steering/kia_soul_ps/steering_control_module.ino [hunk 4]
```diff
@@ -473,6 +516,7 @@ static void check_rx_timeouts( void )
         if( current_ctrl_state.control_enabled ) 
         {
             disableControl();
+            Serial.println("timeout");
```
Comment (LucasBuckland) [inline firmware/steering/kia_soul_ps/steering_control_module.ino]: Debug instead?

**Comment by ltbj** [comment]
This branch and PR are now deprecated in favor of this branch: `add/disable-control-on-driver-input`.

https://github.com/PolySync/OSCC/tree/add/disable-control-on-driver-input

The new implementation uses an exponential filter to calculate the difference in the input torque voltage.

**Pull request closed without merging**